Extensions/xkit_main.js

```diff
@@ -175,6 +175,18 @@
 
 			try {
 				eval(xkit_main.script + "\n//# sourceURL=xkit/" + extension_id + ".js");
+				if (document.location.href.indexOf("www.tumblr.com") === -1 && XKit.extensions[extension_id].onblog !== true) {
+					// Doesn't want to be run on blog page, quit.
+					if (XKit.extensions.xkit_main.disabled_extensions === "") {
+						XKit.extensions.xkit_main.disabled_extensions = extension_id + "(on blog)";
+					} else {
+						XKit.extensions.xkit_main.disabled_extensions = XKit.extensions.xkit_main.disabled_extensions + ", " + extension_id + "(on blog)";
+					}
+					if (!dont_run_next) {
```

Comment:
doesn't this mean that the first non-onblog extension will abort all other extensions?

Reply:
it shouldn't - the code is stolen from the run-or-not-in-iframe logic, and it worked fine when i tested it myself
since xkit preferences isn't onblog i'm sure i would've noticed retags not booting

Reply:
yeah, just take a look at line 137's block - it's the exact same logic